### PR TITLE
Replace getTypeName() with PHP's get_debug_type().

### PR DESCRIPTION
**Pull request by ADmad**

### CODE REVIEW

**Comment by ADmad** [comment]
I dropped `getTypeName()` altogether since replacing it with `get_debug_type()` would be trivial using a search/replace.

File: src/Core/functions.php [hunk 1]
```diff
@@ -312,16 +312,3 @@ function deprecationWarning(string $message, int $stackFrame = 1): void
...
 }
-
-if (!function_exists('getTypeName')) {
-    /**
-     * Returns the objects class or var type of it's not an object
-     *
-     * @param mixed $var Variable to check
-     * @return string Returns the class name or variable type
-     */
-    function getTypeName(mixed $var): string
```
Comment (markstory) [inline src/Core/functions.php:323]: Nice to see that PHP devs were thinking along the same lines as we were.

**Review by markstory — APPROVED** [review]

File: src/Error/Debugger.php [hunk 2]
```diff
@@ -668,16 +668,19 @@ public static function exportVarAsNodes($var, int $maxDepth = 3): NodeInterface
     protected static function export($var, DebugContext $context): NodeInterface
     {
         $type = static::getType($var);
+
+        if (preg_match('/^resource /', $type)) {
```
Comment (garas) [inline src/Error/Debugger.php:672]:
```suggestion
        if (str_starts_with($type, 'resource ')) {
```
PHP 8

**Comment by othercorey** [comment]
Can we add this to a list of changes to consider for `upgrade`?

File: src/Error/Debugger.php (hunk 2 above)
Comment (ADmad) [reply src/Error/Debugger.php:672]: Thanks, will take a while for it to get the new into the daily vocabulary :)

**Comment by othercorey** [comment]
> Can we add this to a list of changes to consider for `upgrade`?

Added to 5.0 wiki

**Comment by ADmad** [comment]
> Added to 5.0 wiki

Thanks

**Comment by markstory** [comment]
If we keep the changes we want to automate later on. I'm happy to build up the appropriate changes for `rector`.

**Review by markstory — APPROVED** [review]

**Pull request merged**